src/main/resources/webroot/main.js

```diff
@@ -185,17 +192,69 @@ angular
         vm.vertxProject.vertxDependencies = [];
         vm.vertxProject.packageName = "";
         vm.vertxProject.jdkVersion = defaults.jdkVersion;
+
+        vm.disableDependencies(defaults.vertxVersion);
+        refreshAvailableDependencies(defaults.vertxVersion, true);
+      }
+
+      function refreshAvailableDependencies(vertxVersion, initializing) {
+        vm.availableVertxDependencies = availableDependencies(vertxVersion, initializing)
+      }
+
+      function availableDependencies(version, initializing) {
+        // filter dependencies by version
+        var filteredByVersion = vm.stack.flatMap(function (category) {
+          return category.items.filter(function (value) {
+            return !vm.exclusions[version].includes(value.artifactId);
+          });
+        });
+
+        // set total number of available dependencies for given version
+        vm.totalPackagesAvailable = filteredByVersion.length;
+        if (initializing) return filteredByVersion;
+
+        // if the form is already initialized filter also by already selected dependencies so they are not available in typeahead
```

Comment:
Can you rename the `initializing` function parameter to `filterSelected`? It would make the code self-documented.

Reply:
Renamed to _dependenciesSelected_ .